src/components/Sing/ScoreSequencer.vue

```diff
@@ -308,256 +300,329 @@ export default defineComponent({
-          (0 < distanceX ? snapWidth.value : -snapWidth.value);
-        dragDurationCurrentX.value = dragDurationCurrentXNext;
-        dragId.value = requestAnimationFrame(dragRight);
-        if (editedNotes.length !== 0) {
-          store.dispatch("UPDATE_NOTES", { notes: editedNotes });
-        }
+      if (previewMode === "RESIZE_RIGHT") {
+        previewResizeRight();
       }
-      dragId.value = requestAnimationFrame(dragRight);
-    };
-
-    // ノート右ドラッグ開始
-    const handleDragRightStart = (event: MouseEvent) => {
-      dragMode.value = DragMode.NOTE_RIGHT;
-      setTimeout(() => {
-        dragDurationCurrentX.value = event.clientX;
-        dragId.value = requestAnimationFrame(dragRight);
-      }, 360);
+      if (previewMode === "RESIZE_LEFT") {
+        previewResizeLeft();
+      }
+      previewRequestId = requestAnimationFrame(preview);
     };
 
-    // ノート左ドラッグ
-    // FIXME: 左右ドラッグロジックを統一する
-    const dragLeft = () => {
-      if (dragMode.value !== DragMode.NOTE_LEFT) {
-        cancelAnimationFrame(dragId.value);
-        return;
-      }
-      const distanceX = cursorX.value - dragDurationCurrentX.value;
-      if (snapWidth.value <= Math.abs(distanceX)) {
-        const editedNotes: Note[] = [];
-        for (const note of selectedNotes.value) {
-          const position =
-            note.position +
-            (0 < distanceX ? snapTicks.value : -snapTicks.value);
-          const duration =
-            note.duration +
-            (0 > distanceX ? snapTicks.value : -snapTicks.value);
-          if (duration < Math.max(snapTicks.value, 0) || note.position < 0) {
-            continue;
-          }
-          editedNotes.push({ ...note, position, duration });
-        }
-        const dragDurationCurrentXNext =
-          dragDurationCurrentX.value +
-          (0 < distanceX ? snapWidth.value : -snapWidth.value);
-        dragDurationCurrentX.value = dragDurationCurrentXNext;
-        dragId.value = requestAnimationFrame(dragLeft);
-        if (editedNotes.length !== 0) {
-          store.dispatch("UPDATE_NOTES", { notes: editedNotes });
-        }
-      }
-      dragId.value = requestAnimationFrame(dragLeft);
+    const getXInBorderBox = (clientX: number, element: HTMLElement) => {
+      return clientX - element.getBoundingClientRect().left;
     };
 
-    // ノート左ドラッグ開始
-    const handleDragLeftStart = (event: MouseEvent) => {
-      dragMode.value = DragMode.NOTE_LEFT;
-      setTimeout(() => {
-        dragDurationCurrentX.value = event.clientX;
-        dragId.value = requestAnimationFrame(dragLeft);
-      }, 360);
+    const getYInBorderBox = (clientY: number, element: HTMLElement) => {
+      return clientY - element.getBoundingClientRect().top;
     };
 
-    // X軸ズーム
-    const setZoomX = (event: Event) => {
+    const startPreview = (
+      event: MouseEvent,
+      mode: PreviewMode,
+      note?: Note
+    ) => {
+      if (nowPreviewing.value) {
+        return;
+      }
```

Comment:
たぶんノート端をクリックしたときに、RESIZEとADDが連続して呼ばれちゃうのを防止するreturnですよね！
エラーの握り潰しになっちゃっているかなと思いました。

連続して呼ばれるのを直接防止する場合は、ノートのonClickのとこでイベントのバブリングを停止する`stopPropagation`を呼ぶのが想定に合ってるかもです。
ただイベントのバブリングは僕もベストプラクティスがわかっておらず、それが良い手法なのかは自信がないです･･･。

とりあえず意図をコメントしつつ、想定外だったらthrowする形はどうでしょう。
```suggestion
      if (nowPreviewing.value) {
        // RESIZE_RIGHT・RESIZE_LEFTのあとにADDも発生するので、その場合は無視する
        if (
          mode === "ADD" &&
          (previewMode === "RESIZE_RIGHT" || previewMode === "RESIZE_LEFT")
        ) {
          return;
        } else {
          throw new Error("nowPreviewing is true.");
        }
      }
```

Reply:
シーケンサーの外でMouseUpが発生した場合、プレビュー状態のままになるので、その時も（MouseDownした場合に）このreturnのところに来ます！
確実にMouseUpイベントが発生するか分からないので（この辺り自身がなかったので）とりあえずreturnする形にしています…！

Reply:
あ～～～なーるほどです！！　意図と違ったコメント書いてしまってすみません！！ 🙇